src/riak_kv_sweeper_cli.erl

```diff
@@ -0,0 +1,180 @@
+register_cli() ->
+    register_all_usage(),
+    register_all_commands().
+
+register_all_usage() ->
+    clique:register_usage(["riak-admin", "sweeper"], sweeper_usage()),
+    clique:register_usage(["riak-admin", "sweeper", "status"], status_usage()).
+
+register_all_commands() ->
+    lists:foreach(fun(Args) -> apply(clique, register_command, Args) end,
+                  [status_register()]).
+
+status_register() ->
+    [["riak-admin", "sweeper", "status"], % Cmd
+     [],                                  % KeySpecs
+     [],                                  % FlagSpecs
+     fun status/3].                       % Implementation callback.
+
+sweeper_usage() ->
+    [
+     "riak-admin sweeper <sub-command>\n\n",
+     "  Sub-commands:\n",
+     "    status           Display sweeper status\n",
+     "  Use --help after a sub-command for more details.\n"
+    ].
+
+status_usage() ->
+    ["riak-admin sweeper status\n\n",
+     "  Display sweeper status\n"].
+
+status(_CmdBase, [], []) ->
+    {Participants, Sweeps} = riak_kv_sweeper:status(),
+    SortedParticipants = lists:keysort(1, Participants),
+    IndexedParticipants =
+        lists:zip(lists:seq(1, length(Participants)), SortedParticipants),
+    ParticipantsTable = format_participants(IndexedParticipants),
+    SweepsTable = format_sweeps(Sweeps, IndexedParticipants),
+    ActiveSweeps = format_active_sweeps(Sweeps, IndexedParticipants),
+    ParticipantsTable ++ SweepsTable ++ ActiveSweeps.
+
+format_sweeps(Sweeps, IndexedParticipants) ->
+    Now = os:timestamp(),
+    Header = io_lib:format("~s", ["Sweep results:"]),
+    SortedSweeps = lists:keysort(#sweep.index, Sweeps),
+    Rows =
+        [begin
+             StartTime = Sweep#sweep.start_time,
+             EndTime = Sweep#sweep.end_time,
+             LastSweep = format_timestamp(Now, StartTime),
+             Duration = format_timestamp(EndTime, StartTime),
+             ResultsColumns  =
+                 format_results(Now, Sweep#sweep.results, IndexedParticipants),
+             [{"Index", Sweep#sweep.index},
+              {"Last sweep", LastSweep},
+              {"Duration", Duration}
+             ] ++ ResultsColumns
+         end
+         || Sweep <- SortedSweeps],
+    [clique_status:text(Header),
+     clique_status:table(Rows)].
+
+format_active_sweeps(Sweeps, IndexedParticipants) ->
+    Rows = [format_progress(Sweep, IndexedParticipants)
+              || #sweep{state = State} = Sweep <- Sweeps, State == running],
+    case Rows of
+        [] ->
+            [];
+        Rows ->
+            Header = io_lib:format("~s", ["Running sweeps:"]),
+            [clique_status:text(Header),
+             clique_status:table(Rows)]
+    end.
+
+format_progress(#sweep{index = Index,
+                       active_participants = AcitvePart,
+                       estimated_keys = {EstimatedKeys, _TS},
+                       swept_keys = SweptKeys},
+                IndexedParticipants) ->
+    case EstimatedKeys of
```

Comment:
Thanks, did this a while ago, and just noticed I did not reply to the review comment until now. Done.